FTI installation fails

```
==> Installing fti-1.5-3jmzik7aag7liq7p6vre5c6zcdqfzd4v
==> No binary for fti-1.5-3jmzik7aag7liq7p6vre5c6zcdqfzd4v found: installing from source
==> Fetching https://github.com/leobago/fti/archive/v1.5.tar.gz
==> No patches needed for fti
==> fti: Executing phase: 'cmake'
==> Error: AttributeError: 'Gcc' object has no attribute 'f90'

/opt/spack/var/spack/repos/fti/packages/fti/package.py:66, in cmake_args:
         65    def cmake_args(self):
  >>     66        if '+fortran' in self.spec and (self.compiler.f90 is None) or (self.compiler.fc is None):
         67            raise InstallError('+fortran was selected, however, no Fortran compiler is available!')
         68        args = [
         69            self.define_from_variant('ENABLE_FORTRAN', 'fortran'),

See build log for details:
  /tmp/root/spack-stage/spack-stage-fti-1.5-3jmzik7aag7liq7p6vre5c6zcdqfzd4v/spack-build-out.txt
```

Installation fails with the error above

I'm testing, however I hit a few other issues in the way.
1. FTI fails to detect HDF5,
2. Build fails with clang+gfortran (there is no reason, these compilers are compatible...)

HDF5
```
==> Installing fti-1.6-xwa5zti24n4oaqgwgos45z2uetjuqg2o
==> No binary for fti-1.6-xwa5zti24n4oaqgwgos45z2uetjuqg2o found: installing from source
==> Fetching https://github.com/leobago/fti/archive/v1.6.tar.gz
==> No patches needed for fti
==> fti: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/fti-1.6-xwa5zti24n4oaqgwgos45z2uetjuqg2o' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/fti-1.6-xwa5zti24n4oaqgwgos45z2uetjuqg2o/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/fti-1.6-xwa5zti24n4oaqgwgos45z2uetjuqg2o/lib64;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/hdf5-1.10.8-2iipobriycf6yjpkr5hv2uflsq6mpmlp/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/openmpi-2.1.1-cldpnhfwyrlmblzgbsqwjz2ha4533bw7/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/hwloc-1.11.13-xkhqja7b47hpjityippg6vynolkyjp3w/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libpciaccess-0.16-ryhmw2gqeeqclg65uuabqgsgpiac2nm2/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libxml2-2.9.12-5tibenwe3xcdiyaojsdzedobgj3dczog/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libiconv-1.16-qi7dxj6rgdydno5mdjzyolz6applztig/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/zlib-1.2.11-3rlgy7ycxtoho44una6o3itgfjltkmpd/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/ncurses-6.2-d34lizgt45miuctn4cmx7quklvukdd2w/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/rocm-smi-lib-4.3.1-xpg3jyjbgfoavjsthwl2zqql7xpzjdti/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/numactl-2.0.14-3opot4q2nxdmogm7t5pt5wkhvapqex76/lib' '-DCMAKE_PREFIX_PATH:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/hdf5-1.10.8-2iipobriycf6yjpkr5hv2uflsq6mpmlp;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/openmpi-2.1.1-cldpnhfwyrlmblzgbsqwjz2ha4533bw7;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/numactl-2.0.14-3opot4q2nxdmogm7t5pt5wkhvapqex76;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/hwloc-1.11.13-xkhqja7b47hpjityippg6vynolkyjp3w;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/rocm-smi-lib-4.3.1-xpg3jyjbgfoavjsthwl2zqql7xpzjdti;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libpciaccess-0.16-ryhmw2gqeeqclg65uuabqgsgpiac2nm2;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/cmake-3.21.4-jc24iz64bdpi5xnna4lficw4ze2pkm64;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/ncurses-6.2-d34lizgt45miuctn4cmx7quklvukdd2w;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libxml2-2.9.12-5tibenwe3xcdiyaojsdzedobgj3dczog;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libiconv-1.16-qi7dxj6rgdydno5mdjzyolz6applztig;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/zlib-1.2.11-3rlgy7ycxtoho44una6o3itgfjltkmpd' '-DENABLE_FORTRAN:BOOL=OFF' '-DENABLE_EXAMPLES:BOOL=OFF' '-DENABLE_SIONLIB:BOOL=OFF' '-DENABLE_HDF5:BOOL=ON' '-DENABLE_FI_IO:BOOL=OFF' '-DENABLE_LUSTRE:BOOL=OFF' '-DENABLE_DOCU:BOOL=OFF' '-DENABLE_TUTORIAL:BOOL=OFF' '-DENABLE_IME_NATIVE:BOOL=OFF' '-DENABLE_TESTS:BOOL=OFF' '/tmp/root/spack-stage/spack-stage-fti-1.6-xwa5zti24n4oaqgwgos45z2uetjuqg2o/spack-src'

1 error found in build log:
     14    -- Detecting CXX compile features - done
     15    -- Found MPI: /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.
           0/openmpi-2.1.1-cldpnhfwyrlmblzgbsqwjz2ha4533bw7/bin/mpicc;/opt/spac
           k/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/openmpi-2.1.1-cldpnhf
           wyrlmblzgbsqwjz2ha4533bw7/bin/mpicxx
     16    
     17    -- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libssl.so
     18    
     19    -- Found ZLIB: /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5
           .0/zlib-1.2.11-3rlgy7ycxtoho44una6o3itgfjltkmpd/lib/libz.so (found v
           ersion "1.2.11")
  >> 20    CMake Error at /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5
           .0/cmake-3.21.4-jc24iz64bdpi5xnna4lficw4ze2pkm64/share/cmake-3.21/Mo
           dules/FindPackageHandleStandardArgs.cmake:230 (message):
     21      Could NOT find HDF5 (missing: HDF5_HL_LIBRARIES)
     22    Call Stack (most recent call first):
     23      /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/cmake-3.21
           .4-jc24iz64bdpi5xnna4lficw4ze2pkm64/share/cmake-3.21/Modules/FindPac
           kageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
     24      /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/cmake-3.21
           .4-jc24iz64bdpi5xnna4lficw4ze2pkm64/share/cmake-3.21/Modules/FindHDF
           5.cmake:1001 (find_package_handle_standard_args)
     25      CMakeLists.txt:106 (find_package)
     26    

See build log for details:
  /tmp/root/spack-stage/spack-stage-fti-1.6-xwa5zti24n4oaqgwgos45z2uetjuqg2o/spack-build-out.txt
```

Clang+GFortran
```
==> Installing fti-1.6-jv7safanjzjc5xsjj7wjca7kib4wnh36
==> No binary for fti-1.6-jv7safanjzjc5xsjj7wjca7kib4wnh36 found: installing from source
==> Warning: microarchitecture specific optimizations are not supported yet on mixed compiler toolchains [check clang@8.0.0 for further details]
==> Fetching https://github.com/leobago/fti/archive/v1.6.tar.gz
==> No patches needed for fti
==> fti: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/fti-1.6-jv7safanjzjc5xsjj7wjca7kib4wnh36' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/fti-1.6-jv7safanjzjc5xsjj7wjca7kib4wnh36/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/fti-1.6-jv7safanjzjc5xsjj7wjca7kib4wnh36/lib64;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/openmpi-2.1.1-xotkq6dsf32gaxx3nrws3eylbav5lobv/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/hwloc-1.11.13-fokswfrwqmmg4uhk72abiqmdov2nee7c/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/libpciaccess-0.16-zcrrgtb7uiavighucqqt7a6ly62kvpxl/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/libxml2-2.9.12-djvyx6nezx33oz4rkshhfuh6hvzzpx3z/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/libiconv-1.16-i77dkfahqgdjwn63xrteeot6oetgbicn/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/zlib-1.2.11-ett7kkkldueryst4vcuq2ms77ixhnhau/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/ncurses-6.2-3gtaftsl4rsorqxc5gss5e32j6iuefz4/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/rocm-smi-lib-4.3.1-avnavzpkwhisbchjf3leqzsykqie6stt/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/numactl-2.0.14-wf7zwlbs4ochqcz7sflzlggdksrpeipi/lib' '-DCMAKE_PREFIX_PATH:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/openmpi-2.1.1-xotkq6dsf32gaxx3nrws3eylbav5lobv;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/numactl-2.0.14-wf7zwlbs4ochqcz7sflzlggdksrpeipi;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/hwloc-1.11.13-fokswfrwqmmg4uhk72abiqmdov2nee7c;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/rocm-smi-lib-4.3.1-avnavzpkwhisbchjf3leqzsykqie6stt;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/libpciaccess-0.16-zcrrgtb7uiavighucqqt7a6ly62kvpxl;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/cmake-3.21.4-ozfo6uebqxxrdaku7tjdspvrpq4oc65f;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/ncurses-6.2-3gtaftsl4rsorqxc5gss5e32j6iuefz4;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/libxml2-2.9.12-djvyx6nezx33oz4rkshhfuh6hvzzpx3z;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/libiconv-1.16-i77dkfahqgdjwn63xrteeot6oetgbicn;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8.0.0/zlib-1.2.11-ett7kkkldueryst4vcuq2ms77ixhnhau' '-DENABLE_FORTRAN:BOOL=ON' '-DENABLE_EXAMPLES:BOOL=OFF' '-DENABLE_SIONLIB:BOOL=OFF' '-DENABLE_HDF5:BOOL=OFF' '-DENABLE_FI_IO:BOOL=OFF' '-DENABLE_LUSTRE:BOOL=OFF' '-DENABLE_DOCU:BOOL=OFF' '-DENABLE_TUTORIAL:BOOL=OFF' '-DENABLE_IME_NATIVE:BOOL=OFF' '-DENABLE_TESTS:BOOL=OFF' '/tmp/root/spack-stage/spack-stage-fti-1.6-jv7safanjzjc5xsjj7wjca7kib4wnh36/spack-src'

1 error found in build log:
     26    
     27    -- Found ZLIB: /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-8
           .0.0/zlib-1.2.11-ett7kkkldueryst4vcuq2ms77ixhnhau/lib/libz.so (found
            version "1.2.11")
     28    -- Detecting Fortran/C Interface
     29    -- Detecting Fortran/C Interface - Found GLOBAL and MODULE mangling
     30    -- Performing Test FC_HAS_CPP_FLAG
     31    -- Performing Test FC_HAS_CPP_FLAG - Success
  >> 32    CMake Error at CMakeLists.txt:215 (message):
     33      Different compiler idetifications for Fortran and C!
     34    
     35    
     36    -- Configuring incomplete, errors occurred!
     37    See also "/tmp/root/spack-stage/spack-stage-fti-1.6-jv7safanjzjc5xsj
           j7wjca7kib4wnh36/spack-build-jv7safa/CMakeFiles/CMakeOutput.log".

See build log for details:
  /tmp/root/spack-stage/spack-stage-fti-1.6-jv7safanjzjc5xsjj7wjca7kib4wnh36/spack-build-out.txt
```

--- a/packages/fti/package.py
+++ b/packages/fti/package.py
@@ -63,7 +63,7 @@ class Fti(CMakePackage):
     depends_on('doxygen',                       type='build', when='+doc')
 
     def cmake_args(self):
-        if '+fortran' in self.spec and (self.compiler.f90 is None) or (self.compiler.fc is None):
+        if '+fortran' in self.spec and (self.compiler.fc is None) and (self.compiler.f90 is None) and (self.compiler.f77 is None):
             raise InstallError('+fortran was selected, however, no Fortran compiler is available!')
         args = [
             self.define_from_variant('ENABLE_FORTRAN', 'fortran'),

